link/src/test/java/com/stripe/android/link/ui/signup/SignUpViewModelTest.kt

```diff
@@ -0,0 +1,450 @@
+    private val dispatcher = UnconfinedTestDispatcher()
+    private val config = LinkConfiguration(
+        stripeIntent = StripeIntentFixtures.PI_SUCCEEDED,
+        merchantName = MERCHANT_NAME,
+        merchantCountryCode = "",
+        customerInfo = LinkConfiguration.CustomerInfo(
+            name = CUSTOMER_NAME,
+            email = CUSTOMER_EMAIL,
+            phone = CUSTOMER_PHONE,
+            billingCountryCode = CUSTOMER_BILLING_COUNTRY_CODE
+        ),
+        shippingValues = null,
+        flags = emptyMap(),
+        cardBrandChoice = null,
+        passthroughModeEnabled = false
+    )
+    private val defaultArgs = LinkActivityContract.Args(
+        configuration = config,
+    )
+
+    private val navController: NavHostController = mock()
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(dispatcher)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `init sends analytics event`() = runTest(dispatcher) {
+        val linkEventsReporter = object : SignUpLinkEventsReporter() {
+            override fun onSignupFlowPresented() {
+                calledCount += 1
+            }
+        }
+
+        createViewModel(linkEventsReporter = linkEventsReporter)
+
+        assertThat(linkEventsReporter.calledCount).isEqualTo(1)
+    }
+
+    @Test
+    fun `When email is valid then lookup is triggered with delay`() = runTest(dispatcher) {
+        val linkAccountManager = object : FakeLinkAccountManager() {
+            override suspend fun lookupConsumer(email: String, startSession: Boolean): Result<LinkAccount?> {
+                delay(100)
+                return super.lookupConsumer(email, startSession)
+            }
+        }
+        val viewModel = createViewModel(prefilledEmail = null, linkAccountManager = linkAccountManager)
+
+        assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.InputtingPrimaryField)
+
+        viewModel.emailController.onRawValueChange("valid@email.com")
+        assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.InputtingPrimaryField)
+
+        // Advance past lookup debounce delay
+        advanceTimeBy(SignUpViewModel.LOOKUP_DEBOUNCE + 1.milliseconds)
+
+        assertThat(viewModel.emailController.fieldValue.value).isEqualTo("valid@email.com")
+        assertThat(viewModel.contentState.signUpState).isEqualTo(SignUpState.VerifyingEmail)
+    }
+
+    @Test
+    fun `When multiple valid emails entered quickly then lookup is triggered only for last one`() =
+        runTest(dispatcher) {
+            val linkAccountManager = object : FakeLinkAccountManager() {
+                var calledCount = 0
+                val emails = arrayListOf<String>()
+                override suspend fun lookupConsumer(email: String, startSession: Boolean): Result<LinkAccount?> {
+                    calledCount += 1
+                    emails.add(email)
+                    delay(100)
```

Comment:
(just clarifying) why do we need the delay here? I thought that the delay was handled within the VM, so we wouldn't also need one here